File: CS12BM Spring19/pa3/DictionaryTest.java
public class DictionaryTest{
	public static void main (String[] args){
		String v;
		Dictionary D = new Dictionary();
		//isEmpty() test
		System.out.println("Empty? " + D.isEmpty());
		//size() test
		System.out.println("Size:" + D.size());
		//insert() test
		D.insert("1","a");
		D.insert("2","b");
		D.insert("3","c");	
		//lookup & error testing 
		v = D.lookup("1");
		System.out.println("key=1 "+(v==null?"not found":("value="+v)));
		v = D.lookup("3");
		System.out.println("key=3 "+(v==null?"not found":("value="+v)));
		v = D.lookup("7");
		System.out.println("key=7 "+(v==null?"not found":("value="+v)));
		//delete() test
		D.delete("1");
		D.delete("3");
		System.out.println(D);
	}
}

File: CS12BM Spring19/lab5/DictionaryTest.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){
	Dictionary A = newDictionary();
	char* k;
	char* v;
	char* word1[] = {"one","two","three","four"};
	char* word2[] = {"test1","test2","test3","test4"};
	int i;

	for(i=0; i<4; i++){
		insert(A, word1[i], word2[i]);
	}

	printDictionary(stdout, A);

	for(i=0; i<4; i++){
		k = word1[i];
		v = lookup(A, k);
		printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
	}

	delete(A, "1");
	delete(A, "3");

	printDictionary(stdout, A);

	for(i=0; i<4; i++){
		k = word1[i];
		v = lookup(A, k);
		printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
	}

	printf("%s\n", (isEmpty(A)?"true":"false"));
	printf("%d\n", size(A));
	makeEmpty(A);
	printf("%s\n", (isEmpty(A)?"true":"false"));

	freeDictionary(&A);

	return(EXIT_SUCCESS);
}